### PR TITLE
BF(TST): make tests use _path_ helper for Windows "friendliness" of the tests

### LINKED ISSUE #2564: `path_is_subpath()` is broken on Windows
```
[00:15:30] ======================================================================
[00:15:30] FAIL: datalad.tests.test_utils.test_path_is_subpath
[00:15:30] ----------------------------------------------------------------------
[00:15:30] Traceback (most recent call last):
[00:15:30]   File "C:\Miniconda35\envs\test-environment\lib\site-packages\nose\case.py", line 198, in runTest
[00:15:30]     self.test(*self.arg)
[00:15:30]   File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\tests\test_utils.py", line 775, in test_path_is_subpath
[00:15:30]     ok_(path_is_subpath('/a/b', '/a'))
[00:15:30] AssertionError
```

### PR DESCRIPTION
**Pull request by yarikoptic**
b91b494b89 simply annotated those tests which were failing on windows to be skipped without
considering to fix them. Those tests were failing because hardcoded POSIX path separator (/).
Quick solution to have tests fixed is to use _path_ helper around those POSIX paths.
The underlying tested code AFAIK should be ok - CI is yet to prove or disprove it.

Closes #2564

### CODE REVIEW

**Comment by yarikoptic** [comment]
uff. codespell became "smarter" - we need to account for that (separate PR)

**Comment by yarikoptic** [comment]
ha -- current fails look legit -- I will try to update my windows VM license in coming days to troubleshoot:
```
______________________________ test_p_startswith ______________________________
    def test_p_startswith():
        ok_(path_startswith(_p('/a/b'), _p('/a')))
        ok_(path_startswith(_p('/a/b'), _p('/a/b')))
        ok_(path_startswith(_p('/a/b'), _p('/a/b/')))
        ok_(path_startswith(_p('/a/b/'), _p('/a/b')))
        ok_(path_startswith(_p('/a/b'), _p('/')))
        ok_(path_startswith(_p('/aaa/b/c'), _p('/aaa')))
>       nok_(path_startswith(_p('/aaa/b/c'), _p('/aa')))
..\datalad\tests\test_utils.py:945: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
expr = True, msg = None
    def assert_false(expr, msg=None):
        if msg is None:
>           assert not expr
E           assert not True
..\datalad\tests\utils_pytest.py:113: AssertionError
______________________________ test_p_is_subpath ______________________________
    def test_p_is_subpath():
>       ok_(path_is_subpath(_p('/a/b'), _p('/a')))
..\datalad\tests\test_utils.py:954: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
expr = False, msg = None
    def assert_true(expr, msg=None):
        if msg is None:
>           assert expr
E           assert False
..\datalad\tests\utils_pytest.py:191: AssertionError
============================== warnings summary ===============================
datalad/distributed/tests/test_ria_basics.py::test_initremote_basic_httpsurl
```

**Comment by adswa** [comment]
I can replicate the failure on my windows machine

File: datalad/tests/test_utils.py [hunk 1]
```diff
@@ -923,37 +923,47 @@ def test_get_dataset_root(path=None):
...
-    nok_(path_startswith('/aaa/b/c', '/aa'))
-    nok_(path_startswith('/a/b', '/a/c'))
-    nok_(path_startswith('/a/b/c', '/a/c'))
+def _p(p: str) -> str:
+    """A helper to code paths as POSIX paths in tests  below. Would prepend fake drive
+       C: to absolute paths on Windows"""
+    if on_windows:
+        pm = p.replace('/', os.sep)
+        if p.startswith('/'):
+            return "C:{pm}"
```
Comment (adswa) [inline datalad/tests/test_utils.py:932]: I think its just a typo?

```suggestion
            return f"C:{pm}"
```

Comment (adswa) [reply datalad/tests/test_utils.py:932]: tests pass with that fix

Comment (yarikoptic) [reply datalad/tests/test_utils.py:932]: oh hoh hoh, Thank you @adswa!

**Comment by yarikoptic** [comment]
Thank you @adswa !!! It is as green as it can get ATM, so ready for merge ;)

File: datalad/tests/test_utils.py [hunk 2]
```diff
@@ -923,37 +923,47 @@ def test_get_dataset_root(path=None):
...
+        else:
+            return pm
+    return p
+
+
 def test_path_startswith():
```
Comment (yarikoptic) [inline datalad/tests/test_utils.py:938]: had to fix up accidentally mass renamed `_path_ -> _p_` here ;)  tests should pass and then I will merge -- no code changes, just got tests to pass on exotic Windows. What could go wrong?

File: datalad/tests/test_utils.py (hunk 1 above)
Comment (adswa) [reply datalad/tests/test_utils.py:932]: I think you resolved this comment, but the fix hasn't yet made it into a commit

Comment (yarikoptic) [reply datalad/tests/test_utils.py:932]: it made it! But then I managed to force push local fix up of a name which I rewrote but without your commit!

Let's commit it again , and thanks again!!!

**Review by adswa — APPROVED** [review]
Now its green! :)

**Pull request merged**